sklearn/preprocessing/_encoders.py

```diff
@@ -778,8 +791,9 @@ def fit(self, X, y=None):
         """
         # base classes uses _categories to deal with deprecations in
         # OneHoteEncoder: can be removed once deprecations are removed
+
         self._categories = self.categories
-        self._fit(X)
+        self._fit(X, encode_missing=self.encode_missing, force_all_finite=True)
```

Comment:
Why `force_all_finite=True` here and not False? I would think that in the OrdinalEncoder we now allow NaNs?